src/coreclr/src/jit/emitxarch.cpp

```diff
@@ -214,6 +214,65 @@ bool emitter::AreUpper32BitsZero(regNumber reg)
     return false;
 }
 
+//------------------------------------------------------------------------
+// IsZeroFlagSet: Checks if the previous instruction set the zero flag for
+//                reg of opSize size to zero.
+//
+// Arguments:
+//    reg - register of interest
+//    opSize - size of register
+//
+// Return Value:
+//    true if the previous instruction set the zero flag for reg
+//    false if not, or if we can't safely determine
+//
+// Notes:
+//    Currently only looks back one instruction.
+bool emitter::IsZeroFlagSet(regNumber reg, emitAttr opSize)
+{
+    // Don't look back across IG boundaries (possible control flow)
+    if (emitCurIGinsCnt == 0)
```

Comment:
>emitCurIGinsCnt [](start = 8, length = 15)

As @andya@microsoft.com  pointed out, check for extended IG using something like `bool isFirstInstrInBlock = (emitCurIGinsCnt == 0) && ((emitCurIG->igFlags & IGF_EXTEND) == 0);`